src/main/java/org/itsallcode/holidays/calculator/logic/variants/Holiday.java

```diff
@@ -77,21 +117,33 @@ public boolean equals(Object obj) {
 			return false;
 		}
 		final Holiday other = (Holiday) obj;
+		if (alternative == null) {
```

Comment:
You could use [equalsverifier](https://jqno.nl/equalsverifier/) to test `equals` and `hashCode` methods.

Reply:
Hello @kaklakariada did you add your comment to the right line of code? 
As far as I understood: equalsverifier is for testing, while the file Holiday.java is main code.
Or did I get something wrong?

Reply:
I added the comment not to this specific line, just a general remark 😅

Reply:
I get a lot of strange errors that seem hard to fix. 
After playing around a little bit with equalsverifier I find it too difficult to use and the effort compared to the benefit too high.